### PR TITLE
 browsing by directories and tags

### LINKED ISSUE #3: Directory Browsing only works two directories down
Browsing directories doesn't work multiple directories down. I've tested this using Ario (GTK frontend to mpd) and ncmpc.

For instance, the directory `John Lennon/Imagine` would be browsable, however `John Lennon/Studio Albums/Imagine` is not.

### PR DESCRIPTION
**Pull request by hhm0**
This pull allows browsing more than two directories down, and also allows browsing by artist/album and by album.

Fixes #3 .

### CODE REVIEW

File: mopidy_subidy/library.py [hunk 1]
```diff
@@ -31,13 +37,32 @@ def lookup_artist(self, artist_id):
         return self.subsonic_api.get_artist_by_id(artist_id)
 
     def browse(self, browse_uri):
-        type = uri.get_type(browse_uri)
+        vdir_templates = [
+            # ("root", <no name>)
```
Comment (Prior99) [inline mopidy_subidy/library.py]: Is this an unneeded line or documentation? Please remove it or elaborate (I know that there is very little documentation present in the project)

File: mopidy_subidy/library.py [hunk 2]
```diff
@@ -31,13 +37,32 @@ def lookup_artist(self, artist_id):
...
+            ("rootdirs", "Directories"),
+            ("artists", "Artists"),
+            ("albums", "Albums"),
+        ]
+        vdirs = dict((template[0], dict(id=template[0], name=template[1])) for template in vdir_templates)
```
Comment (Prior99) [inline mopidy_subidy/library.py]: Could you make this multiline to make it more readable?

**Comment by hhm0** [comment]
I refactored the code; I hope it is clearer now. :-)

**Comment by Prior99** [comment]
When browsing a directory with multiple levels I am getting the following error:

```
ERROR    SubidyBackend backend caused an exception.
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mopidy/core/library.py", line 19, in _backend_error_handling
    yield
  File "/usr/lib/python2.7/site-packages/mopidy/core/library.py", line 112, in _browse
    result = backend.library.browse(uri).get()
  File "build/bdist.linux-x86_64/egg/pykka/threading.py", line 52, in get
    compat.reraise(*self._data['exc_info'])
  File "build/bdist.linux-x86_64/egg/pykka/compat.py", line 12, in reraise
    exec('raise tp, value, tb')
  File "build/bdist.linux-x86_64/egg/pykka/actor.py", line 201, in _actor_loop
    response = self._handle_receive(message)
  File "build/bdist.linux-x86_64/egg/pykka/actor.py", line 295, in _handle_receive
    return callee(*message['args'], **message['kwargs'])
  File "/home/prior/repositories/mopidy-subidy/mopidy_subidy/library.py", line 82, in browse
    return self.browse_diritems(uri.get_directory_id(browse_uri))
  File "/home/prior/repositories/mopidy-subidy/mopidy_subidy/library.py", line 56, in browse_diritems
    return self.subsonic_api.get_diritems_as_refs(directory_id)
  File "/home/prior/repositories/mopidy-subidy/mopidy_subidy/subsonic_api.py", line 262, in get_diritems_as_refs
    return [(self.raw_directory_to_ref(diritem) if diritem.get('isDir') else self.raw_song_to_ref(diritem)) for diritem in self.get_raw_dir(directory_id)]
  File "/home/prior/repositories/mopidy-subidy/mopidy_subidy/subsonic_api.py", line 197, in get_raw_dir
    return sorted(diritems, key=diritem_sort_key)
  File "/home/prior/repositories/mopidy-subidy/mopidy_subidy/subsonic_api.py", line 35, in diritem_sort_key
    key = int(item['track'])
KeyError: 'track'
```

Let's see if I can fix that myself, though.

**Comment by Prior99** [comment]
I guess that file didn't have a track. Let us just use `1` as a default if a file has no track number.

**Pull request merged**